src/FakeItEasy/Core/FakeManager.cs

```diff
@@ -48,10 +48,7 @@ internal FakeManager(Type fakeObjectType, object proxy, string? fakeObjectName)
             this.FakeObjectType = fakeObjectType;
             this.FakeObjectName = fakeObjectName;
 
-            this.preUserRules = new IFakeObjectCallRule[]
-            {
-                new EventRule(this),
-            };
+            this.postUserRules = SharedPostUserRules.Prepend<IFakeObjectCallRule>(new EventRule(this)).ToList();

```

Comment:
I worry about the effect this will have on performance. We spent a fair amount of effort teasing apart constructed lists of rules during the big performance overhaul. I'm not going to die on this hill or anything, but maybe it's worth considering or investigating. I'm not entirely sure how expensive prepending, to-listing, and then switching to enumerating over the list is. It's less readable, but maybe we could've just renamed the preuser rules (and former postuser, why not?) and shifted the iteration lower, especially since right now our per-instance postuser rules are all grouped together?

Reply:
> I'm not entirely sure how expensive prepending, to-listing, and then switching to enumerating over the list is

Prepend is quite cheap, I think. ToList is a bit more expensive, and maybe not necessary. Enumeration is probably not very expensive.

> but maybe we could've just renamed the preuser rules (and former postuser, why not?) and shifted the iteration lower, especially since right now our per-instance postuser rules are all grouped together?

I'm not sure I follow. Could you elaborate?

Reply:
Ah, I see. In fact, since there's only one rule in `individualPostUserRules`, maybe we could just remove the array and directly call the rule; even better as far as performance is concerned, but maybe not as clean.

Reply:
Done